### ISSUE TITLE
Last saved checkpoint not resumed when using trainer.fit(..., ckpt_path='last')

### ISSUE THREAD

**Issue by mlgavilan** [Lightning-AI/lightning#14466]
## 🐛 Bug

Last saved checkpoint (by setting save_last=True in ModelCheckpoint callback) is not automatically resumed when specifying ckpt_path='last' argument in trainer.fit method. 

From the related code (https://github.com/Lightning-AI/lightning/blob/1.7.3/src/pytorch_lightning/trainer/trainer.py#L1419), I see that the last model path is taken from the "last_model_path" attribute of the ModelCheckpoint callback object, but its value is always an empty string at the moment of resuming the latest checkpoint.

One workaround is to manually set this attribute just after creating the callback, but I don't know if this was the intended behaviour or if there is another (and possibly better) way to do all this.

### To Reproduce

Run this code a first time and stop it after some epochs have been completed. Then run again. The second run does not resume the latest saved checkpoint but starts the training again from the first epoch.

```python
import os

import torch
from torch.utils.data import DataLoader, Dataset
from time import sleep
from pytorch_lightning import LightningModule, Trainer
from pytorch_lightning.callbacks import ModelCheckpoint


class RandomDataset(Dataset):
    def __init__(self, size, length):
        self.len = length
        self.data = torch.randn(length, size)

    def __getitem__(self, index):
        return self.data[index]

    def __len__(self):
        return self.len


class BoringModel(LightningModule):
    def __init__(self):
        super().__init__()
        self.layer = torch.nn.Linear(32, 2)

    def forward(self, x):
        return self.layer(x)

    def training_step(self, batch, batch_idx):
        loss = self(batch).sum()
        self.log("train_loss", loss)
        return {"loss": loss}

    def validation_step(self, batch, batch_idx):
        loss = self(batch).sum()
        self.log("valid_loss", loss)

    def on_validation_epoch_end(self) -> None:
        sleep(1)

    def configure_optimizers(self):
        return torch.optim.SGD(self.layer.parameters(), lr=0.1)


def run():

    train_data = DataLoader(RandomDataset(32, 64), batch_size=2)
    val_data = DataLoader(RandomDataset(32, 64), batch_size=2)

    model = BoringModel()
    trainer = Trainer(
        callbacks=ModelCheckpoint(monitor='valid_loss', mode='min', filename='bestLoss', save_last=True),
        accelerator='cpu',
        logger=None,
        default_root_dir=os.getcwd(),
        max_epochs=1000)

    # fit
    trainer.fit(model, train_dataloaders=train_data, val_dataloaders=val_data, ckpt_path='last')


if __name__ == "__main__":
    run()

```

### Expected behavior

Resume training from the last saved checkpoint when using ckpt_path="last" option in trainer.fit

### Environment
<details>
  <summary>Details</summary>

        * CUDA:
                - GPU:
                        - Quadro T1000
                - available:         True
                - version:           10.2
        * Lightning:
                - pytorch-lightning: 1.7.3
                - torch:             1.12.1
                - torchmetrics:      0.9.3
        * Packages:
                - absl-py:           1.2.0
                - aiohttp:           3.8.1
                - aiosignal:         1.2.0
                - async-timeout:     4.0.2
                - attrs:             22.1.0
                - cachetools:        5.2.0
                - certifi:           2022.6.15
                - charset-normalizer: 2.1.1
                - frozenlist:        1.3.1
                - fsspec:            2022.7.1
                - google-auth:       2.11.0
                - google-auth-oauthlib: 0.4.6
                - grpcio:            1.48.0
                - idna:              3.3
                - importlib-metadata: 4.12.0
                - markdown:          3.4.1
                - markupsafe:        2.1.1
                - multidict:         6.0.2
                - numpy:             1.23.2
                - oauthlib:          3.2.0
                - packaging:         21.3
                - pip:               22.2.2
                - protobuf:          3.19.4
                - pyasn1:            0.4.8
                - pyasn1-modules:    0.2.8
                - pydeprecate:       0.3.2
                - pyparsing:         3.0.9
                - pytorch-lightning: 1.7.3
                - pyyaml:            6.0
                - requests:          2.28.1
                - requests-oauthlib: 1.3.1
                - rsa:               4.9
                - setuptools:        63.3.0
                - six:               1.16.0
                - tensorboard:       2.10.0
                - tensorboard-data-server: 0.6.1
                - tensorboard-plugin-wit: 1.8.1
                - torch:             1.12.1
                - torchmetrics:      0.9.3
                - tqdm:              4.64.0
                - typing-extensions: 4.3.0
                - urllib3:           1.26.12
                - werkzeug:          2.2.2
                - wheel:             0.37.1
                - yarl:              1.8.1
                - zipp:              3.8.1
        * System:
                - OS:                Linux
                - architecture:
                        - 64bit
                        - ELF
                - processor:         x86_64
                - python:            3.8.10
                - version:           #49~20.04.1-Ubuntu SMP Thu Aug 4 19:15:44 UTC 2022
</details>

### Additional context

cc @justusschock @awaelchli @ninginthecloud @rohitgr7 @otaj @akihironitta

**Comment by otaj** [comment]
Hi @mlgavilan, I just wanted to let you know, that it is being worked on and I have not abandoned this issue. Though I don't think I have an ETA for the fix sadly yet.

**Comment by otaj** [comment]
Hi @mlgavilan, I pushed a PR that should fix this. Please let me know if this fixes your issue - I tested it locally, but maybe you had something else in mind.
